userscript/src/hooks/useOnRemove.ts

```diff
@@ -0,0 +1,19 @@
+import { useMutationObserver } from './useMutationObserver';
+
+export function useOnRemove(element: Element, callback: () => void) {
+  useMutationObserver(
+    element?.parentElement,
+    (mutations, observer) => {
+      for (const mutation of mutations) {
+        const removedNodes = Array.from(mutation.removedNodes);
+        for (const el of removedNodes) {
+          if (el === element) {
+            callback();
+            observer.disconnect();
+          }
+        }
+      }
+    },
+    { childList: true },
+  );
```

Comment:
Conditional call of react hooks is prohibited. The `useMutationObserver` hook is capable of handling undefined elements, in which cases it will not observe anything.